tools/bazar/controllers/EntryController.php

```diff
@@ -75,15 +94,14 @@ public function view($entryId, $time = '', $showFooter = true)
         // If not found, use default template
         if (empty($renderedEntry)) {
             for ($i = 0; $i < count($form['template']); ++$i) {
-                if ($form['prepared'][$i] instanceof BazarField) {
-                    $field = $form['prepared'][$i];
-                    if ($this->wiki->CheckACL($field->getReadAccess(), null, true, $entryId)) {
+                // Check if we should display the field
+                if (empty($form['template'][$i][11]) ||
```

Comment:
Juste en lisant cette proposition, c'est pas très clair sur pourquoi les nouvelles lignes de code fonctionne, et pas l'ancienne. C'est quoi le raisonnement ?

Avec les différents retours à la ligne, j'ai du mal à distinguer l'imbrication des if/else :-/

Reply:
c'est bien pile ici la modif, tous les autres changements c'est du refacto
ligne 80 : `if ($this->wiki->CheckACL($field->getReadAccess(), null, true, $entryId)) {`
il fallait remplacer par `(empty(...) || $this->wiki->CheckACL..)`
j'ai juste ensuite remonté le test car le bon test se faisait plus bas dans le cas où le champ n'était pas un BazarField (ancienne méthode où on appelle une fonction)

Je suis d'accord pour la lisibilité, j'avais indenté différemment mais vu qu'on passe tout en PSR mon éditeur a reformaté derrière. Si t'as une solution qui garde le format PSR, je suis preneur :)